conans/client/proxy.py

```diff
@@ -96,6 +99,8 @@ def get_package(self, package_ref, short_paths):
             output.success('Already installed!')
             installed = True
             log_package_got_from_local_cache(package_ref)
+            if self._recorder:
```

Comment:
this idiom all over the place ``if self._recorder`` seems like could be improved

Reply:
Yes, I will pass always the recorder from the copy and uploader managers.